src/core_plugins/kibana/ui_setting_defaults.js

```diff
@@ -226,6 +226,12 @@ export function getUiSettingDefaults() {
       value: '($0,0.[00])',
       description: 'Default <a href="http://numeraljs.com/" target="_blank">numeral format</a> for the "currency" format'
     },
+    'format:language': {
```

Comment:
:thumbsup: to making this setting more specific. I'm also slightly concerned about how this is a global setting. Perhaps `format:number:defaultLanguage` will keep it in line with `format:number:defaultPattern` and leaves room for a per-field language setting should we want to add one down the road.

Reply:
Hi,

`format:number:defaultLanguage` looks a good idea. 
That said, few points about that :
- first, this setting will apply on every type format concerned by numeraljs, so : number, bytes, percent, and currency. Not only number.
- then, currently (and since a long time), numeraljs allows only to define a global static language, not one for each formatter.
- would it make sense to have a dashboard with different numeric fields formatted with different languages ?
- in the future, an interesting evolution could be to detect browser language ?

=> Given these points, what setting name do you choose ? :)

Reply:
> numeraljs allows only to define a global static language

ugh, bummer. Is it unrealistic to call numeral.language() before every conversion?

> would it make sense to have a dashboard with different numeric fields formatted with different languages?

The numeral language defines the currency symbol right? If so I think it makes sense to have a unique language per field if two fields represent different currencies.

Reply:
Looks like `numeral.language()` is just setting an internal value inside the library, so if we called it before calling `numeralInst.set().format(this.param('pattern'))` we could use a different language for each field.

Reply:
@fbaligand I would probably move forward with `format:number:defaultLanguage` in this pr and do the per-field values in another pr, another day.

Reply:
If we do move forward with applying this language to all instances of `numeral`, it'd be more consistent to do so in the manner described in this [comment](https://github.com/elastic/kibana/pull/14019#discussion_r139674713)

Reply:
@spalger I agree with you, it's fine to have `defaultLanguage` here to be able to add per-field value later.
That said, in original numeraljs project (https://github.com/adamwdraper/Numeral-js), the last version use the "locale" term instead of "language" term.
So question is : do you prefer `format:number:defaultLanguage` or `format:number:defaultLocale`

Reply:
I'm not too concerned about updates in numeral; I don't think we'll ever upgrade because of it's breaking changes and how that would impact everyone who has saved formats already. It's more likely that we will do something like we have done with kuery and offer a choice between legacy numeral and some custom solution we could build in the future.

That said, I think that locale is more accurate description of the choice, so I would go with `format:number:defaultLocale`

Reply:
OK, for this new name, I will commit it.

Reply:
@spalger 
new name 'format:number:defaultLocale' is commited